src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -984,6 +984,11 @@ private void checkout(GitSCM scm, Run<?,?> build, GitClient git, TaskListener li
     public SCMRevision getTrustedRevision(SCMRevision revision, TaskListener listener)
             throws IOException, InterruptedException {
         if (revision instanceof PullRequestSCMRevision) {
+            PullRequestSCMHead head = (PullRequestSCMHead) revision.getHead();
+            if (repoOwner.equals(head.getSourceRepo()) && repository.equals(head.getSourceRepo())) {
```

Comment:
I think you meant

```java
if (repoOwner.equals(head.getSourceOwner()) && repository.equals(head.getSourceRepo())) {
```

or even

```java
if (repoOwner.equals(head.getSourceOwner())) {
```

since GH would not allow the repository name to be distinct.